src/legacy/core_plugins/expressions/public/np_ready/public/expression_renderer.tsx

```diff
@@ -70,5 +93,37 @@ export const createRenderer = (loader: IExpressionLoader): ExpressionRenderer =>
     mountpoint.current,
   ]);
 
-  return <div {...dataAttrs} className={className} ref={mountpoint} />;
+  useEffect(() => {
+    // We only want a clean up to run when the entire component is unloaded, not on every render
+    return function cleanup() {
+      if (handlerRef.current) {
+        handlerRef.current.destroy();
+        handlerRef.current = null;
+      }
+    };
+  }, []);
+
+  const classes = classNames('expExpressionRenderer', className, {
+    'expExpressionRenderer-isLoading': state.isLoading,
+    'expExpressionRenderer-hasError': state.hasError,
+  });
+
+  return (
+    <div {...dataAttrs} className={classes}>
+      {state.isLoading ? (
+        <>
+          <EuiProgress size="xs" color="accent" position="absolute" />
+          <EuiLoadingChart mono size="l" />
+        </>
+      ) : null}
+      {!state.isLoading && state.hasError ? (
+        options.renderError ? (
+          options.renderError(state.errorMessage)
```

Comment:
It seems I'm always getting a flash of the error message. Is there anyway to prevent this? 

![ezgif-1-abf796548c06](https://user-images.githubusercontent.com/549577/67505715-c5a5a400-f659-11e9-925a-e659c6ab6ceb.gif)